graph-subscriptions-api/datasource/src/datsource_postgres.rs

```diff
@@ -176,26 +191,34 @@ impl Datasource for DatasourcePostgres {
         let limit = first.unwrap_or(100);
         let offset = skip.unwrap_or(0);
 
+        // **NOTE**: `COALESCE(JSON_AGG(result.ticket_payload) ->> 0, '{}')`
+        // The `ticket_payload` is a JSON object stored in the DB.
+        // This select aggregates all of the `result.ticket_payload` data into a JSON array,
+        // because this value could potentially become _very, very_ large,
+        // we want to just return the first item from the array.
+        // For most use-cases, this is fine as each entry in the array will be the same,
+        // and this value is used to "reconstruct" the ticket in a UI to let a user resign the
+        // request ticket message with the same domain to get the same value.
```

Comment:
I'm learning

Reply:
Initially, when I pushed this branch, I just used `JSON_AGG`. Which in my local testing was fine. But had an epiphany last night of like, if a user makes 100 queries on this ticket, this field would return a JSON array with 100 entries, and that feels really inefficient and bad. So decided to dig further into it and came across this.